go.work

```diff
@@ -1,4 +1,6 @@
-go 1.20
+go 1.21
```

Comment:
https://github.com/googleapis/google-cloud-go?tab=readme-ov-file#go-versions-supported 

Minimum version currently is go 1.20. Please do not merge this PR.

Reply:
Minimum version is go 1.21 now